### PR TITLE
Sg/ui ux sticky note

### PR DESCRIPTION
**Pull request by martint-unity**
### Purpose of this PR
Added sticky notes to Shader graph.

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?automation-tools_branch=master&ScriptableRenderLoop_branch=sg%2Fui-ux-sticky-note&unity_branch=trunk

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low

**Halo Effect**: None, Low, Medium, High?
Low

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by spaztronaut** [comment]
87a3e06ff91a6edc1dcaccf96d354105
cant copy and paste sticky nodes. there should be an option to do so in the context menu. also, am able to "cut" when the sticky node is in a multiselection with other nodes but when pasting the clipboard the sticky node is not added to the graph

**Comment by spaztronaut** [comment]
2870887f059d1d67d171bd4dbd5a337d
i think that the sticky node should be added to the subgraph if part of the convert to subgraph selection

**Comment by spaztronaut** [comment]
there are a lot of other issues but those are also present in VFX graph so i wont include them here

**Comment by martint-unity** [comment]
@spaztronaut added functionality for adding into subgraphs

**Comment by spaztronaut** [comment]
> 2870887f059d1d67d171bd4dbd5a337d
> i think that the sticky node should be added to the subgraph if it is part of the convert to subgraph selection

looks good now 👍 

@pbbastian you can look at the code

**Comment by martint-unity** [comment]
We are waiting for @pbbastian :)

File: com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+    {
+        void OnStartResize();
+        void OnResized();
+    }
+
+    class ResizableElementFactory : UxmlFactory<ResizableElement>
+    {}
+    class ElementResizer : Manipulator
+    {
+//        static MethodInfo m_ValidateLayoutMethod;
```
Comment (pbbastian) [inline com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs]: remove commented code

unless it was just copied like this from VFX

File: com.unity.shadergraph/Editor/Drawing/Views/GraphEditorView.cs [hunk 2]
```diff
@@ -488,28 +513,44 @@ public void HandleGraphChanges()
...
+                GraphElement graphElement = null;
+                if (groupChange.groupItem is AbstractMaterialNode node)
                 {
-                    var groupView = nodeView.GetContainingScope() as ShaderGroup;
+                    graphElement = m_GraphView.GetNodeByGuid(node.guid.ToString());
+                }
+                else if (groupChange.groupItem is StickyNoteData stickyNote)
+                {
+                    graphElement = m_GraphView.GetElementByGuid(stickyNote.guid.ToString());
+                }
```
Comment (pbbastian) [inline com.unity.shadergraph/Editor/Drawing/Views/GraphEditorView.cs]: add an `else` clause that throws a `InvalidOperationException` with an appropriate message

File: com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+                        (resizedTarget as IResizable).OnResized();
+                    }
+                }
+                target.UnregisterCallback<MouseMoveEvent>(OnMouseMove);
+                target.ReleaseMouse();
+                e.StopPropagation();
+            }
+        }
+    }
+    public class ResizableElement : VisualElement
```
Comment (pbbastian) [inline com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs]: should not be `public`

File: com.unity.shadergraph/Editor/Util/CopyPasteGraph.cs [hunk 4]
```diff
@@ -3,6 +3,7 @@
 using System.Linq;
 using UnityEngine;
 using UnityEditor.ShaderGraph;
+using UnityEditor.ShaderGraph.Drawing;
```
Comment (pbbastian) [inline com.unity.shadergraph/Editor/Util/CopyPasteGraph.cs]: this doesn't seem like it's needed

File: com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs (hunk 1 above)
Comment (martint-unity) [reply com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs]: Copied like this from VFX

File: com.unity.shadergraph/Editor/Drawing/Views/GraphEditorView.cs [hunk 5]
```diff
@@ -495,28 +520,48 @@ public void HandleGraphChanges()
...
+                    graphElement = m_GraphView.GetNodeByGuid(node.guid.ToString());
+                }
+                else if (groupChange.groupItem is StickyNoteData stickyNote)
                 {
-                    var groupView = nodeView.GetContainingScope() as ShaderGroup;
+                    graphElement = m_GraphView.GetElementByGuid(stickyNote.guid.ToString());
+                }
+                else
+                {
+                    throw new InvalidOperationException("Couldn't find any graph elements.");
```
Comment (pbbastian) [inline com.unity.shadergraph/Editor/Drawing/Views/GraphEditorView.cs]:
```suggestion
                    throw new InvalidOperationException("Unknown group item type.");
```

**Pull request merged**